src/django/api/serializers/facility/utils.py

```diff
@@ -450,3 +450,57 @@ def depromote_unspecified_items(items: list):
         k['values'][0] == 'Unspecified' and
         len(k['values']) == 1),
         reverse=False)
+
+
+def format_sectors(items,
```

Comment:
Nice idea to move this functionality to a separate function. I think we could apply more refactoring techniques to this function.

```
def format_sectors(items, claims, date_field_to_sort, use_main_created_at, user_can_see_detail):
    def is_contributor_visible(item, is_claim):
        if is_claim:
                 return user_can_see_detail
        return (user_can_see_detail and item['source']['is_active']
                and item['source']['is_public'] and item['has_active_complete_match'])

    def format_sector_data(entity, is_claim):
        created_at = format_date(entity['created_at']) if use_main_created_at else None
        return {
            'created_at': created_at,
            'updated_at': format_date(entity['updated_at']),
            'contributor_id': get_contributor_id_from_facilityindexnew(entity['contributor'],
                                                                       is_contributor_visible(entity, is_claim)),
            'contributor_name': get_contributor_name_from_facilityindexnew(entity['contributor'],
                                                                           is_contributor_visible(entity, is_claim)),
            'values': entity['sector'],
            'is_from_claim': is_claim
        }

    item_sectors = [format_sector_data(item, False) for item in items]
    claim_sectors = [format_sector_data(claim, True) for claim in claims]

    sorted_item_sectors = sorted(item_sectors, key=lambda i: i[date_field_to_sort], reverse=True)

    return claim_sectors + depromote_unspecified_items(sorted_item_sectors)
```